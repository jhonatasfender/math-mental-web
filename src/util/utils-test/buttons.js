import { screen } from '@testing-library/react';

export const buttons = () => {
  const [
    seven,
    eight,
    nine,
    four,
    five,
    six,
    one,
    two,
    three,
    zero,
    point,
    clear,
  ] = screen.getAllByRole('button');

  return {
    0: zero,
    1: one,
    2: two,
    3: three,
    4: four,
    5: five,
    6: six,
    7: seven,
    8: eight,
    9: nine,
    ',': point,
    C: clear,
  };
};
