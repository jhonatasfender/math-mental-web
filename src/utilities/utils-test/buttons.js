import { screen } from '@testing-library/react';

export const buttons = () => ({
  0: screen.getByRole('button', { name: '0' }),
  1: screen.getByRole('button', { name: '1' }),
  2: screen.getByRole('button', { name: '2' }),
  3: screen.getByRole('button', { name: '3' }),
  4: screen.getByRole('button', { name: '4' }),
  5: screen.getByRole('button', { name: '5' }),
  6: screen.getByRole('button', { name: '6' }),
  7: screen.getByRole('button', { name: '7' }),
  8: screen.getByRole('button', { name: '8' }),
  9: screen.getByRole('button', { name: '9' }),
  ',': screen.getByRole('button', { name: '1' }),
  C: screen.getByRole('button', { name: 'c' }),
});
