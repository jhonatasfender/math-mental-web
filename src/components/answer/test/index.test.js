import Answer from '@components/answer';
import Node from '@components/latex';
import { fireEvent, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { buttons } from '@utils-test/buttons';

describe('<Answer />', () => {
  const renderer = (viewing) => {
    const all = render(
      <Answer
        viewing={
          viewing || (() => <Node aria-label="operation">10 + 10 = 20</Node>)
        }
      />,
    );
    return all;
  };

  it('should to render', () => {
    const { container } = renderer();
    expect(container).toMatchSnapshot();
  });

  it('should to have the numbers', () => {
    renderer();
    Object.values(buttons()).forEach((el) => {
      expect(el).toBeInTheDocument();
    });
  });

  it('should render the number when clicking the button', () => {
    let validExpect = '';
    const viewing = (result) => {
      expect(validExpect).toEqual(result);
    };

    renderer(viewing);

    Object.values(buttons()).forEach((el) => {
      validExpect = el.textContent === 'C' ? '' : validExpect + el.textContent;
      userEvent.click(el);
    });
  });

  it('should render the number when typing on the keyboard', () => {
    let validExpect = '';
    const viewing = (result) => {
      expect(validExpect).toEqual(result);
    };

    const { container } = renderer(viewing);

    const keyDown = (keyboard) => {
      validExpect =
        keyboard.key === 'c' || keyboard.key === 'Backspace'
          ? ''
          : validExpect + keyboard.key;
      fireEvent.keyDown(container, keyboard);
    };

    const typing = () => {
      keyDown({ key: '1', code: 'Digit1', keyCode: 49 });
      keyDown({ key: '2', code: 'Digit2', keyCode: 50 });
      keyDown({ key: '3', code: 'Digit3', keyCode: 51 });
      keyDown({ key: '4', code: 'Digit4', keyCode: 52 });
      keyDown({ key: '5', code: 'Digit5', keyCode: 53 });
      keyDown({ key: '6', code: 'Digit6', keyCode: 54 });
      keyDown({ key: '7', code: 'Digit7', keyCode: 55 });
      keyDown({ key: '8', code: 'Digit8', keyCode: 56 });
      keyDown({ key: '9', code: 'Digit9', keyCode: 57 });
      keyDown({ key: '0', code: 'Digit0', keyCode: 48 });
      keyDown({ key: ',', code: 'Comma', keyCode: 188 });
    };

    typing();
    keyDown({ key: 'c', code: 'KeyC', keyCode: 67, charCode: 0 });

    typing();
    keyDown({ key: 'Backspace', code: 'Backspace', keyCode: 8, charCode: 0 });
  });
});
