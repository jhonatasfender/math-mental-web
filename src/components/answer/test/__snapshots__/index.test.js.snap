// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Answer /> should to render 1`] = `
<div>
  <div
    class="answer__Container-iybhpq-1"
  >
    <div
      aria-label="operation"
      class="latex__Container-sc-1gi2td7-0 drmSQN"
    >
      <span
        class="katex"
      >
        <span
          aria-hidden="true"
          class="katex-html"
        >
          <span
            class="base"
          >
            <span
              class="strut"
              style="height:0.72777em;vertical-align:-0.08333em;"
            />
            <span
              class="mord"
            >
              10
            </span>
            <span
              class="mspace"
              style="margin-right:0.2222222222222222em;"
            />
            <span
              class="mbin"
            >
              +
            </span>
            <span
              class="mspace"
              style="margin-right:0.2222222222222222em;"
            />
          </span>
          <span
            class="base"
          >
            <span
              class="strut"
              style="height:0.64444em;vertical-align:0em;"
            />
            <span
              class="mord"
            >
              10
            </span>
            <span
              class="mspace"
              style="margin-right:0.2777777777777778em;"
            />
            <span
              class="mrel"
            >
              =
            </span>
            <span
              class="mspace"
              style="margin-right:0.2777777777777778em;"
            />
          </span>
          <span
            class="base"
          >
            <span
              class="strut"
              style="height:0.64444em;vertical-align:0em;"
            />
            <span
              class="mord"
            >
              20
            </span>
          </span>
        </span>
      </span>
    </div>
    <div
      class="answer__Buttons-iybhpq-0 oQXPE"
    >
      <button
        aria-label="7"
        class="button__Btn-usgv6-0 jxsgGL"
        type="button"
      >
        <div
          class="latex__Container-sc-1gi2td7-0 drmSQN"
        >
          <span
            class="katex"
          >
            <span
              aria-hidden="true"
              class="katex-html"
            >
              <span
                class="base"
              >
                <span
                  class="strut"
                  style="height:0.64444em;vertical-align:0em;"
                />
                <span
                  class="mord"
                >
                  7
                </span>
              </span>
            </span>
          </span>
        </div>
      </button>
      <button
        aria-label="8"
        class="button__Btn-usgv6-0 jxsgGL"
        type="button"
      >
        <div
          class="latex__Container-sc-1gi2td7-0 drmSQN"
        >
          <span
            class="katex"
          >
            <span
              aria-hidden="true"
              class="katex-html"
            >
              <span
                class="base"
              >
                <span
                  class="strut"
                  style="height:0.64444em;vertical-align:0em;"
                />
                <span
                  class="mord"
                >
                  8
                </span>
              </span>
            </span>
          </span>
        </div>
      </button>
      <button
        aria-label="9"
        class="button__Btn-usgv6-0 jxsgGL"
        type="button"
      >
        <div
          class="latex__Container-sc-1gi2td7-0 drmSQN"
        >
          <span
            class="katex"
          >
            <span
              aria-hidden="true"
              class="katex-html"
            >
              <span
                class="base"
              >
                <span
                  class="strut"
                  style="height:0.64444em;vertical-align:0em;"
                />
                <span
                  class="mord"
                >
                  9
                </span>
              </span>
            </span>
          </span>
        </div>
      </button>
      <button
        aria-label="4"
        class="button__Btn-usgv6-0 jxsgGL"
        type="button"
      >
        <div
          class="latex__Container-sc-1gi2td7-0 drmSQN"
        >
          <span
            class="katex"
          >
            <span
              aria-hidden="true"
              class="katex-html"
            >
              <span
                class="base"
              >
                <span
                  class="strut"
                  style="height:0.64444em;vertical-align:0em;"
                />
                <span
                  class="mord"
                >
                  4
                </span>
              </span>
            </span>
          </span>
        </div>
      </button>
      <button
        aria-label="5"
        class="button__Btn-usgv6-0 jxsgGL"
        type="button"
      >
        <div
          class="latex__Container-sc-1gi2td7-0 drmSQN"
        >
          <span
            class="katex"
          >
            <span
              aria-hidden="true"
              class="katex-html"
            >
              <span
                class="base"
              >
                <span
                  class="strut"
                  style="height:0.64444em;vertical-align:0em;"
                />
                <span
                  class="mord"
                >
                  5
                </span>
              </span>
            </span>
          </span>
        </div>
      </button>
      <button
        aria-label="6"
        class="button__Btn-usgv6-0 jxsgGL"
        type="button"
      >
        <div
          class="latex__Container-sc-1gi2td7-0 drmSQN"
        >
          <span
            class="katex"
          >
            <span
              aria-hidden="true"
              class="katex-html"
            >
              <span
                class="base"
              >
                <span
                  class="strut"
                  style="height:0.64444em;vertical-align:0em;"
                />
                <span
                  class="mord"
                >
                  6
                </span>
              </span>
            </span>
          </span>
        </div>
      </button>
      <button
        aria-label="1"
        class="button__Btn-usgv6-0 jxsgGL"
        type="button"
      >
        <div
          class="latex__Container-sc-1gi2td7-0 drmSQN"
        >
          <span
            class="katex"
          >
            <span
              aria-hidden="true"
              class="katex-html"
            >
              <span
                class="base"
              >
                <span
                  class="strut"
                  style="height:0.64444em;vertical-align:0em;"
                />
                <span
                  class="mord"
                >
                  1
                </span>
              </span>
            </span>
          </span>
        </div>
      </button>
      <button
        aria-label="2"
        class="button__Btn-usgv6-0 jxsgGL"
        type="button"
      >
        <div
          class="latex__Container-sc-1gi2td7-0 drmSQN"
        >
          <span
            class="katex"
          >
            <span
              aria-hidden="true"
              class="katex-html"
            >
              <span
                class="base"
              >
                <span
                  class="strut"
                  style="height:0.64444em;vertical-align:0em;"
                />
                <span
                  class="mord"
                >
                  2
                </span>
              </span>
            </span>
          </span>
        </div>
      </button>
      <button
        aria-label="3"
        class="button__Btn-usgv6-0 jxsgGL"
        type="button"
      >
        <div
          class="latex__Container-sc-1gi2td7-0 drmSQN"
        >
          <span
            class="katex"
          >
            <span
              aria-hidden="true"
              class="katex-html"
            >
              <span
                class="base"
              >
                <span
                  class="strut"
                  style="height:0.64444em;vertical-align:0em;"
                />
                <span
                  class="mord"
                >
                  3
                </span>
              </span>
            </span>
          </span>
        </div>
      </button>
      <button
        aria-label="0"
        class="button__Btn-usgv6-0 jxsgGL"
        type="button"
      >
        <div
          class="latex__Container-sc-1gi2td7-0 drmSQN"
        >
          <span
            class="katex"
          >
            <span
              aria-hidden="true"
              class="katex-html"
            >
              <span
                class="base"
              >
                <span
                  class="strut"
                  style="height:0.64444em;vertical-align:0em;"
                />
                <span
                  class="mord"
                >
                  0
                </span>
              </span>
            </span>
          </span>
        </div>
      </button>
      <button
        aria-label=","
        class="button__Btn-usgv6-0 jxsgGL"
        type="button"
      >
        <div
          class="latex__Container-sc-1gi2td7-0 drmSQN"
        >
          <span
            class="katex"
          >
            <span
              aria-hidden="true"
              class="katex-html"
            >
              <span
                class="base"
              >
                <span
                  class="strut"
                  style="height:0.3em;vertical-align:-0.19444em;"
                />
                <span
                  class="mpunct"
                >
                  ,
                </span>
              </span>
            </span>
          </span>
        </div>
      </button>
      <button
        aria-label="c"
        class="button__Btn-usgv6-0 jxsgGL"
        type="button"
      >
        <div
          class="latex__Container-sc-1gi2td7-0 drmSQN"
        >
          <span
            class="katex"
          >
            <span
              aria-hidden="true"
              class="katex-html"
            >
              <span
                class="base"
              >
                <span
                  class="strut"
                  style="height:0.68333em;vertical-align:0em;"
                />
                <span
                  class="mord mathnormal"
                  style="margin-right:0.07153em;"
                >
                  C
                </span>
              </span>
            </span>
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
